client/rest/src/main/java/org/elasticsearch/client/Response.java

```diff
@@ -124,9 +124,9 @@ public HttpEntity getEntity() {
             final Matcher matcher = WARNING_HEADER_PATTERN.matcher(warning);
             if (matcher.matches()) {
                 warnings.add(matcher.group(1));
-                continue;
+            } else {
```

Comment:
I flipped this around because it lines up a bit better with the rest of our style.